alibi_detect/base.py

```diff
@@ -122,6 +122,9 @@ def drift_config(self):
         if self.preprocess_fn is not None:
             cfg.update({'preprocess_fn': self.preprocess_fn})
 
+        # Add version_warning if it exists
+        cfg.update({'version_warning': self.meta.get('version_warning', False)})
```

Comment:
If a `version_warning` was raised by `validate_config`, due to a `version` or `config_spec` mismatch, it is written to the config here. This ensures a warning will still be raised when the newly saved detector is reloaded again.